bundles/org.dataflowanalysis.analysis.dfd/src/org/dataflowanalysis/analysis/dfd/core/DFDFlowGraphCollection.java

```diff
@@ -57,6 +68,19 @@ public List<? extends AbstractTransposeFlowGraph> findTransposeFlowGraphs() {
             logger.error("Cannot find transpose flow graphs for non-dfd resource provider");
             throw new IllegalArgumentException();
         }
-        return new DFDTransposeFlowGraphFinder(dfdResourceProvider).findTransposeFlowGraphs();
+
+        if (transposeFlowGraphFinderClass.equals(DFDTransposeFlowGraphFinder.class))
+            this.transposeFlowGraphFinder = new DFDTransposeFlowGraphFinder(dfdResourceProvider);
+        else
+            this.transposeFlowGraphFinder = new DFDCyclicTransposeFlowGraphFinder(dfdResourceProvider);
+
```

Comment:
This is no longer needed, if an instance of the transpose flow graph finder is provided